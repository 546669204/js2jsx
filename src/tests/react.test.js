const jsToJsxByReact  = require("../react").default;


test("react component jsx",()=>{
  expect(jsToJsxByReact(`t.default = function () {
    const e = o.a.createElement("h1", null, "Hello, ", "Josh Perez");
    return o.a.createElement("div", null, o.a.createElement("button", {
      onClick: this.handleClick.bind(this),
      disabled: this.state.disabled
    }, "Add Item"), this.state.if ? o.a.createElement("div", null, "123456") : null, o.a.createElement("ul", null, this.state.items.map((function (e, t) {
      return o.a.createElement("li", {
        key: t
      }, e)
    }))), e)
  }`)).toMatchSnapshot()
})


