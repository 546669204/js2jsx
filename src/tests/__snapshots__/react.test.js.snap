// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react component jsx 1`] = `
"t.default = function () {
  const e = <h1>Hello, Josh Perez</h1>;
  return <div>{<button onClick={this.handleClick.bind(this)} disabled={this.state.disabled}>Add Item</button>}{this.state.if ? <div>123456</div> : null}{<ul>{this.state.items.map(function (e, t) {
        return <li key={t}>{e}</li>;
      })}</ul>}{e}</div>;
};"
`;
